### PR TITLE
Trunk 4455

### PR DESCRIPTION
**Pull request by kdaud**
https://issues.openmrs.org/browse/TRUNK-4455

### CODE REVIEW

**Comment by georgejunior** [comment]
@kdaud Can you add a commit message as advised at https://wiki.openmrs.org/display/docs/Pull+Request+Tips ?

**Comment by georgejunior** [comment]
Can you squash your commits?

**Comment by HerbertYiga** [comment]
@kdaud   you can have a look at this https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by samuelmale** [comment]
Thanks for working on this; fast things first, did you see the travis failure?

File: api/src/main/java/org/openmrs/OpenmrsMetadata.java [hunk 1]
```diff
@@ -1,81 +0,0 @@
...
- * In OpenMRS, we distinguish between data and metadata within our data model. Metadata represent
- * system and descriptive data such as data types &mdash; a relationship type or encounter type.
- * Metadata are generally referenced by clinical data but don't represent patient-specific data
- * themselves. OpenMRS objects that represent metadata should implement this interface.
- * 
- * @see OpenmrsData
- * @see BaseChangeableOpenmrsMetadata
- * @since 1.5
- */
-public interface OpenmrsMetadata extends Auditable, Retireable {
```
Comment (samuelmale) [inline api/src/main/java/org/openmrs/OpenmrsMetadata.java]: why did you delete this file?

what happens to types implementing `OpenmrsMetadata`, I guess you need to re-analyse what's required by the ticket?

**Comment by kdaud** [comment]
Thanks Herbert for this link
https://wiki.openmrs.org/display/docs/Pull+Request+Tips. It of help
and clarification by @samuelmale is of help. Am gona re-analyse what's
required by the ticket and give it a second attempt again

**Comment by kdaud** [comment]
I suggest writing contract tests that can be run against every
implementation of this interface, Could this achieve the desired
objective of the ticket? @samuelmale, @georgejunior

**Comment by samuelmale** [comment]
I'm not sure I understood what you really mean.

**Comment by dkayiwa** [comment]
@kdaud can you squash these commits into one?
